### PR TITLE
Layer/RMS Norm + Quantization Fusion

### PR DESCRIPTION
**Pull request by yaox12**
# What does this PR do ?

Ideally, we should use `LayerNormLinear` to fuse normalization and quantization. But there're some exceptions, e.g., in MLA, we're using separate input layernorm and `Linear` layers. So we provide this option to fuse quantization in LayerNorm/RMSNorm.

Another option is to use a single GEMM for both kv_down/q_down projection in MLA, just like what we're doing for GQA/MHA, so we can do the fusion through TE's `LayernormLinear`. We will wait for the final decision.

## Contribution process

```mermaid
flowchart LR
    A[Pre-checks] --> B[PR Tests]
    subgraph Code Review/Approval
        C1[Expert Review] --> C2[Final Review]
    end
    B --> C1
    C2 --> D[Merge]
```

### Pre-checks

- [ ] I want this PR in a versioned release and have added the appropriate Milestone (e.g., `Core 0.8`)
- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

The following process is enforced via the CODEOWNERS file for changes into `megatron/core`. For changes outside of `megatron/core`, it is up to the PR author whether or not to tag the Final Reviewer team.

<details>
<summary>For MRs into `main` branch</summary>

#### (Step 2): Collect the expert reviewers reviews

1. Attach the `Expert Review` label when your PR is ready for review.
2. GitHub auto-assigns expert reviewers based on your changes. They will get notified and pick up your PR soon.

:warning: Only proceed to the next step once all reviewers have approved, merge-conflict are resolved and the CI is passing.  
Final Review might get declined if these requirements are not fulfilled.

#### (Step 3): Final Review

1. Add `Final Review` label
2. GitHub auto-assigns final reviewers based on your changes. They will get notified and pick up your PR soon.

#### (Optional Step 4): Cherry-pick into release branch

If this PR also needs to be merged into `core_r*` release branches, after this PR has been merged, select `Cherry-pick` to open a new PR into the release branch.

</details>

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### Merging your PR

Any member of [core-adlr](https://github.com/orgs/teams/NVIDIA/core-adlr) and [`core-nemo`](https://github.com/orgs/teams/NVIDIA/core-nemo) will be able to merge your PR.

### CODE REVIEW

File: megatron/core/fusions/fused_layer_norm.py [hunk 1]
```diff
@@ -31,32 +29,22 @@ class FusedLayerNorm(torch.nn.Module):
...
+      maybe_fuse_quantize (bool): Whether to fuse quantize. Added for compatibility with
+                                  Transformer Engine.
     """
 
     def __init__(
         self,
         config: TransformerConfig,
         hidden_size: int,
         eps: float = 1e-5,
-        persist_layer_norm: bool = True,
```
Comment (yaox12) [inline megatron/core/fusions/fused_layer_norm.py:57]: These args were once removed but re-added in https://github.com/NVIDIA/Megatron-LM/commit/67a0e5df1a51461d707bf6609ce44993eaaee545. I thought it was by accident.

**Comment by yaox12** [comment]
Close this PR in favor of https://github.com/NVIDIA/Megatron-LM/pull/3039.
